cime/src/drivers/mct/main/prep_glc_mod.F90

```diff
@@ -196,6 +244,69 @@ subroutine prep_glc_init(infodata, lnd_c2_glc)
 
     end if
 
+    if (glc_present .and. ocn_c2_glc) then
+
+       call seq_comm_getData(CPLID, &
+            mpicom=mpicom_CPLID, iamroot=iamroot_CPLID)
+
+       o2x_ox => component_get_c2x_cx(ocn(1))
+       lsize_o = mct_aVect_lsize(o2x_ox)
+
+       x2g_gx => component_get_x2c_cx(glc(1))
+       lsize_g = mct_aVect_lsize(x2g_gx)
+
+       allocate(o2x_gx(num_inst_ocn))
+       do eoi = 1,num_inst_ocn
+          call mct_aVect_init(o2x_gx(eoi), rList=seq_flds_o2x_fields, lsize=lsize_g)
+          call mct_aVect_zero(o2x_gx(eoi))
+       enddo
+
+       allocate(x2gacc_gx(num_inst_glc))
+       do egi = 1,num_inst_glc
+          call mct_aVect_init(x2gacc_gx(egi), x2g_gx, lsize_g)
+          call mct_aVect_zero(x2gacc_gx(egi))
+       end do
+
+       x2gacc_gx_cnt = 0
+       samegrid_go = .true.
+       if (trim(ocn_gnam) /= trim(glc_gnam)) samegrid_go = .false.
+       if (iamroot_CPLID) then
+          write(logunit,*) ' '
+          write(logunit,F00) 'Initializing mapper_So2g'
+       end if
+       call seq_map_init_rcfile(mapper_So2g, ocn(1), glc(1), &
+       'seq_maps.rc','ocn2glc_smapname:','ocn2glc_smaptype:',samegrid_go, &
+       'mapper_So2g initialization',esmf_map_flag)
+       if (iamroot_CPLID) then
+          write(logunit,*) ' '
+          write(logunit,F00) 'Initializing mapper_Fo2g'
+       end if
+       call seq_map_init_rcfile(mapper_Fo2g, ocn(1), glc(1), &
+       'seq_maps.rc','ocn2glc_fmapname:','ocn2glc_fmaptype:',samegrid_go, &
+       'mapper_Fo2g initialization',esmf_map_flag)
+
+       !Initialize module-level arrays associated with compute_melt_fluxes
+       allocate(oceanTemperature(lsize_g))
+       allocate(oceanSalinity(lsize_g))
+       allocate(oceanHeatTransferVelocity(lsize_g))
+       allocate(oceanSaltTransferVelocity(lsize_g))
+       allocate(interfacePressure(lsize_g))
+       allocate(iceTemperature(lsize_g))
+       allocate(iceTemperatureDistance(lsize_g))
+       allocate(iceFloatingMask(lsize_g))
+       allocate(outInterfaceSalinity(lsize_g))
+       allocate(outInterfaceTemperature(lsize_g))
+       allocate(outFreshwaterFlux(lsize_g))
+       allocate(outOceanHeatFlux(lsize_g))
+       allocate(outIceHeatFlux(lsize_g))
+       ! TODO: Can we allocate these only while used or are we worried about performance hit?
```

Comment:
@rljacob , is it preferable to 1) allocate these once on init and keep the memory allocated for the entire simulation but only have one allocation or 2) allocate/deallocate these at each coupling interval which would have smaller memory footprint but require many allocate operations?

Reply:
@rljacob , @jgfouca , do either of you have a preference for the implementation of allocate/deallocate here?

Reply:
This routine is only called once per coupling step so it probably won't matter for timing.  If those arrays are huge, you should probably allocate once.

Reply:
@rljacob , I don't think they will ever be huge.  Their length is the number of GLC grid cells that overlap with the OCN grid.  But if it doesn't matter much either way, I'll leave the current implementation for simplicity.